tensorflow_datasets/core/file_format_adapter.py

```diff
@@ -386,6 +386,8 @@ def _item_to_tf_feature(item, key_name):
         "FeatureConnector should return a numpy array with the correct dtype "
         "instead of a Python list.".format(key_name)
     )
+  if isinstance(v, six.binary_type):
+    return tf.train.Feature(bytes_list=tf.train.BytesList(value=[v]))
 
```

Comment:
rm the branch below that deals with binary/unicode. handle those separately up front here.
you can use six.binary_type and six.text_type
tf.compat.as_bytes can handle str/unicode/bytes.

Reply:
and please cover the case of a list/tuple of such as well.